substation-diagram/substation-diagram-cgmes/src/main/java/com/powsybl/substationdiagram/cgmes/LayoutToCgmesExtensionsConverter.java

```diff
@@ -0,0 +1,371 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.substationdiagram.cgmes;
+
+import com.powsybl.cgmes.dl.conversion.CgmesDLUtils;
+import com.powsybl.cgmes.iidm.extensions.dl.*;
+import com.powsybl.iidm.network.*;
+import com.powsybl.substationdiagram.SubstationDiagram;
+import com.powsybl.substationdiagram.layout.*;
+import com.powsybl.substationdiagram.library.ComponentLibrary;
+import com.powsybl.substationdiagram.library.ComponentType;
+import com.powsybl.substationdiagram.library.ResourcesComponentLibrary;
+import com.powsybl.substationdiagram.model.*;
+import com.powsybl.substationdiagram.svg.DefaultSubstationDiagramInitialValueProvider;
+import com.powsybl.substationdiagram.svg.DefaultSubstationDiagramStyleProvider;
+import com.powsybl.substationdiagram.svg.SubstationDiagramStyleProvider;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Christian Biasuzzi <christian.biasuzzi@techrain.eu>
+ */
+public class LayoutToCgmesExtensionsConverter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LayoutToCgmesExtensionsConverter.class);
+
+    private static final double OFFSET_MULTIPLIER_X = 2.0;
+
+    private final LayoutParameters lparams;
+    private final ComponentLibrary convergenceComponentLibrary;
```

Comment:
the nodes' coordinates are filled as a byproduct of writesvg() - same way as it is used in other situations, like the substation diagram viewer or the substationdiagramtool ... it's true that in these case the svg is (one of) the output, though

Reply:
@sylvlecl remark is relevant:

The original design was clear, VoltageLayout is responsible for finding nodes coordinates and SVGwriter only write SVG based on coordinates found by layout step. Unfortunatly it seems that recently added SubstationLayout do not follow this design and that VoltageLevelGraph coordinates (which are the nodes of a SubstationGraph) are assigned in the SVGWriter (see writegraph method). SubstationLayout seem to only be responsible for "snake" lines intermediate point calculation but not for node coordinate calculation.

This is a serious design/impl flaw that force you to use SVGWriter where you should not.

I think that we should not merge this PR as it is as it would "consolidate" a serious design issue in substation layout but we should try to fix it either in this PR or in another one to be merged before this one.

Reply:
Some refactoring of the substation layout design was done in PR 133, refactoring needed to implement the cgmes substation layout (even if, in the code of the PR, the layout is still called by the SVGWriter). To be evaluated if the refactoring goes in the right direction, and what other refactoring could be needed

Reply:
Indeed, I think your refactoring will allow to get rid easily of the use of `writeSvg`, by simply calling `substationLayout.run(...)`.
In this case, it may be easier to first merge #133 .

Reply:
Thanks.   PR #133 (with its refactored substation layout) was merged and the DL converter/exporter has been adapted accordingly; it does not rely on SVGWriter's execution, anymore.